compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/scene/BaseComposeScene.skiko.kt

```diff
@@ -209,7 +223,6 @@ internal abstract class BaseComposeScene(
     private fun doLayout() {
         snapshotInvalidationTracker.onLayout()
         measureAndLayout()
-        inputHandler.onLayout()
```

Comment:
This function was called not only from `render`, so it completely removes `syntheticEventSender.updatePointerPosition()` call from input handling - is it expected?
See `ComposeSceneInputHandler.prepareForPointerInputEvent` usage

Reply:
Yes, we should return sending synthetic events in `prepareForPointerInputEvent` (or in `doLayout`).

The case when that is needed:
1. send scroll event -> handle scroll event-> change scroll position
2. send scroll event -> layout -> send synthetic move caused by the previous scroll -> change hover state that can affect scroll handling -> handle scroll event

As for calling effects - it is enough to call them in `render` only, as the important thing here is to not send an event between recomposition and effect starting.

Reply:
I added `updatePointerPosition()` in `ComposeSceneInputHandler.onPointerEvent`, although I'm not sure about the logic of doing it *before* dispatching the (real) event.

Reply:
It seems like the test cases mentioned by Igor are not covered by our tests. Could you please add a test with it?

Reply:
`MouseMoveTest.hover on scroll in lazy list` seems to cover it. Not sure why it passed.

Reply:
`MouseMoveTest.hover on scroll` and `MouseMoveTest.hover on scroll in lazy list` don't test this because they call `waitForIdle()` after sending the scroll event, which causes the synthetic event to be dispatched in the regular render call.

It appears the scenario @igordmn described doesn't actually work even on `jb-main` because actual scrolling (e.g. `Modifier.verticalScroll`) doesn't happen without recomposition. For example, this test fails on `jb-main` (put it in `MouseMoveTest.kt`):
```
    @Test
    fun scrollEventCausesScrollWithoutRecomposition() = runSkikoComposeUiTest(
        size = Size(100f, 100f),
    ) {
        val collector1 = EventCollector("Collector 1")
        val collector2 = EventCollector("Collector 2")

        setContent {
            val state = rememberScrollState()
            Column(Modifier.size(10.dp).verticalScroll(state)) {
                Box(
                    modifier = Modifier
                        .collectPointerEvents(collector1)
                        .size(10.dp, 10.dp)
                )
                Box(
                    modifier = Modifier
                        .collectPointerEvents(collector2)
                        .size(10.dp, 10.dp)
                )
            }
        }

        scene.sendPointerEvent(PointerEventType.Enter, Offset(0f, 0f))
        collector1.assertCounts(enter = 1, exit = 0)
        collector2.assertCounts(enter = 0, exit = 0)

        scene.sendPointerEvent(PointerEventType.Scroll, Offset(0f, 0f), scrollDelta = Offset(0f, 10000f))
        scene.sendPointerEvent(PointerEventType.Move, Offset(0f, 0f))
        collector1.assertCounts(enter = 1, exit = 1)
        collector2.assertCounts(enter = 1, exit = 0)
    }
```

Reply:
Ok, I guess it's not a blocker